### PR TITLE
refactor: Reorganize multisite handling

### PR DESCRIPTION
**Pull request by rjmackay**
This pull request makes the following changes:
- Replace the old Multisite class with a collection of classes
    - MultisiteManager which handles which site is currently enabled, and switching DB
    - SiteRepo for looking up sites
    - Site to represent the site itself and get urls/config/etc
- Replace DB injection with injecting a Resolver class which gives us the current DB as needed
- Add MultisiteAwareJob class to run jobs in a multisite context
- Updating *all* the repos to use new approach

Test checklist:
- [x] composer test
- [x] I certify that I ran my checklist

Depends on #3372

Fixes ushahidi/platform#2787

Ping @ushahidi/platform

### CODE REVIEW

File: src/App/Multisite/MultisiteManager.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    public function getDomain()
+    {
+        return $this->domain;
+    }
+
+    public function enabled()
+    {
+        return $this->enabled;
```
Comment (kinstelli) [inline src/App/Multisite/MultisiteManager.php]: should we ensure this is boolean?

File: src/App/Multisite/MultisiteManager.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ * @package    Ushahidi\Platform
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Multisite;
+
+use Illuminate\Contracts\Events\Dispatcher;
+
+class MultisiteManager
```
Comment (rjmackay) [inline src/App/Multisite/MultisiteManager.php]: This is probably a good place to start reviewing. This class mostly replaces the old `Multisite` class but split into more sensible pieces

File: src/App/Multisite/Site.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Multisite;
+
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\DB;
+use Ushahidi\Core\Entity\ConfigRepository;
+
+// @todo consider just an Eloquent model? or ushahidi entity
```
Comment (kinstelli) [inline src/App/Multisite/Site.php]: Good question -- I'm not sure how we'd handle the isDBReady stuff if this were an Eloquent model. Split it out?

File: tests/unit/App/Http/Middleware/CheckDemoExpirationTest.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+use Ushahidi\App\Http\Middleware\CheckDemoExpiration;
+use Illuminate\Http\Request;
+use Tests\TestCase;
+use Mockery as M;
+
+/**
+ * @backupGlobals disabled
+ * @preserveGlobalState disabled
+ */
+class CheckDemoExpirationTest extends TestCase
```
Comment (rjmackay) [inline tests/unit/App/Http/Middleware/CheckDemoExpirationTest.php]: Yay more tests

**Pull request merged**